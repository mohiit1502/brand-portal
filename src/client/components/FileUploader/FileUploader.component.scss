@import "../../styles/override/variables.scss";
.c-FileUploader {

}

