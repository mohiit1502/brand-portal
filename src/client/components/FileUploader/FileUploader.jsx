import React from "react";
import PropTypes from "prop-types";
import "./FileUploader.component.scss";
import Tooltip from "../custom-components/tooltip/tooltip";
import * as images from "../../images";
import ProgressBar from "../custom-components/progress-bar/progress-bar";

/* eslint-disable complexity, no-nested-ternary */
const FileUploader = props => {
  const cancelHandler = props.onCancel && props.parentRef[props.onCancel] ? props.cancelHandlerArg ? () => props.parentRef[props.onCancel](props.cancelHandlerArg) : props.parentRef[props.onCancel] : null;
  const tooltipContent = {
    webformDocContent: <div>
      <ol className="m-0 p-0">
        <ul className="m-0 pl-3 text-left font-size-12">
          <li>Please attach supporting documents in .doc,.pdf,.csv format. The file should be less than 7MB</li>
        </ul>
      </ol>
    </div>,
    businessDocContent: <div>
      <ol className="m-0 p-0">
        <ul className="m-0 pl-3 text-left font-size-12">
          <li>Upload an official copy of Business Registration Certificate</li>
          <li>Supported Documents: PDF, DOC, DOCX | Max : 7MB</li>
        </ul>
      </ol>
    </div>,
    additionalDocContent: <div>
      <ol className="m-0 p-0">
        <ul className="m-0 pl-3 text-left font-size-12">
          <li>Upload IP Registration Documents or Letter of Authorization</li>
          <li>Supported Documents: PDF, DOC, DOCX | Max : 7MB</li>
        </ul>
      </ol>
    </div>
  };

  return (
    <div className={`c-FileUploader form-row primary-file-upload mb-3${props.containerClasses ? ` ${  props.containerClasses}` : ""}`}>
      <div className="col">
        <div className="file-upload-title mb-2">
          {props.label} <Tooltip placement={"right"}
            content={tooltipContent[props.tooltipContentKey]}
            icon={images[props.icon]}/>
        </div>
        {
          !props.uploading && !props.id &&
          <>
            <label
              className={`btn btn-sm btn-primary upload-btn my-2${props.disabled ? " disabled" : ""}`}>
              {props.buttonText}
              <input type="file" className="d-none" onChange={props.onChange} onClick={e => e.target.value = null} accept={props.accept}
                disabled={props.disabled}/>
            </label>
            {props.error && <small className="d-block error">{props.error}</small>}
          </>
        }
        {props.uploading && !props.id && <ProgressBar filename={props.filename} uploadPercentage={props.uploadPercentage}/>}
        {!props.uploading && props.id &&
          <div className="col-6 field-container position-relative">
            <div className={`uploaded-file-label form-control mb-2`}>
              <span className="d-block overflow-auto">{props.filename}</span>
            </div>
            <span aria-hidden="true" className="cancel-file-selection-btn position-absolute cursor-pointer"
                onClick={cancelHandler}>&times;</span>
          </div>
        }
      </div>
    </div>
  );
};

FileUploader.propTypes = {
  buttonText: PropTypes.string,
  cancelHandlerArg: PropTypes.string,
  containerClasses: PropTypes.string,
  disabled: PropTypes.bool,
  filename: PropTypes.string,
  icon: PropTypes.string,
  id: PropTypes.string,
  label: PropTypes.string,
  onCancel: PropTypes.func,
  onChange: PropTypes.func,
  parentRef: PropTypes.object,
  setTooltipContent: PropTypes.func,
  tooltipContentKey: PropTypes.string,
  uploading: PropTypes.bool,
  uploadPercentage: PropTypes.number
};

export default FileUploader;
