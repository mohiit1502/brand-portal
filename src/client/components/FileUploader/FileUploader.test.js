/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import FileUploader from "./FileUploader";

describe("FileUploader", () => {
   it("renders without error", () => {

   });
});
