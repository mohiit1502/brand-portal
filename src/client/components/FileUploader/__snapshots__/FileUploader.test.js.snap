// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploader renders without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <FileUploader
    buttonText=""
    cancelHandlerArg=""
    containerClasses=""
    disabled={true}
    filename=""
    icon=""
    id=""
    label=""
    onCancel={[Function]}
    onChange={[Function]}
    parentRef={Object {}}
    setTooltipContent={[Function]}
    tooltipContentKey=""
    uploadPercentage={111}
    uploading={true}
  >
    <div
      className="c-FileUploader form-row primary-file-upload mb-3"
    >
      <div
        className="col"
      >
        <div
          className="file-upload-title mb-2"
        >
           
          <Connect(Tooltip)
            placement="right"
          >
            <Tooltip
              dispatch={[Function]}
              placement="right"
            >
              <img
                alt="i"
                data-html="true"
                data-placement="right"
                data-toggle="tooltip"
                title=""
              />
            </Tooltip>
          </Connect(Tooltip)>
        </div>
        <Connect(ProgressBar)
          filename=""
          uploadPercentage={111}
        >
          <ProgressBar
            dispatch={[Function]}
            filename=""
            uploadPercentage={111}
          >
            <div
              className="progress mb-2 col-6 px-0"
            >
              <div
                className="progress-bar-text d-flex align-items-center h-100"
              >
                Uploading ... 111 %
              </div>
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow={111}
                className="progress-bar progress-bar-animated"
                role="progressbar"
                style={
                  Object {
                    "width": "111%",
                  }
                }
              />
            </div>
          </ProgressBar>
        </Connect(ProgressBar)>
      </div>
    </div>
  </FileUploader>
</Provider>
`;
